docs/api-reference/types/aggregate-root.md

```diff
@@ -88,133 +172,478 @@ public long ExpectedVersion { get; set; }
+{
+    private decimal _balance;
+    private bool _isActive;
+    
+    // Command handler method
+    public void Deposit(decimal amount, string reference, ICorrelatedMessage source)
+    {
+        // Validate business rules
+        if (!_isActive)
+            throw new InvalidOperationException("Cannot deposit to an inactive account");
+            
+        if (amount <= 0)
+            throw new ArgumentException("Amount must be positive", nameof(amount));
+            
+        // Raise the event with proper correlation
+        RaiseEvent(MessageBuilder.From(source, () => new FundsDeposited(
+            Id, 
+            amount, 
+            reference, 
+            DateTime.UtcNow)));
+    }
+    
+    // Event handler method
+    private void Apply(FundsDeposited @event)
+    {
+        _balance += @event.Amount;
+    }
+}
+```
 
-### UpdateWithEvents
+### RestoreFromEvents
 
-Updates this aggregate with the provided events, starting from the expected version.
+Restores this aggregate from the history of events. This method is typically called by the constructor or by a repository when reconstituting an aggregate.
 
 ```csharp
-public void UpdateWithEvents(IEnumerable<object> events, long expectedVersion);
+public void RestoreFromEvents(IEnumerable<object> events);
 ```
 
 **Parameters**:
-- `events` (`System.Collections.Generic.IEnumerable<object>`): The events to update with.
-- `expectedVersion` (`System.Int64`): The expected version to start from.
-
-**Exceptions**:
-- `System.ArgumentNullException`: Thrown when `events` is `null`.
-- `System.InvalidOperationException`: Thrown when this aggregate does not have historical events or expected version mismatch.
+- `events` (`System.Collections.Generic.IEnumerable<object>`): The events to restore from.
 
-**Remarks**: This method checks that the expected version matches the aggregate's current version, and then applies each event in sequence.
+**Example**:
+```csharp
+// Inside a repository implementation
+public TAggregate GetById<TAggregate>(Guid id) where TAggregate : AggregateRoot, new()
+{
+    // Retrieve events from the event store
+    var events = _eventStore.GetEvents(GetStreamName(typeof(TAggregate), id));
+    
+    // Create a new instance of the aggregate
+    var aggregate = new TAggregate();
+    
+    // Set the ID property
+    typeof(TAggregate)
+        .GetProperty("Id", BindingFlags.Public | BindingFlags.Instance)
+        .SetValue(aggregate, id);
+    
+    // Restore the aggregate state from events
+    aggregate.RestoreFromEvents(events);
+    
+    // Set the expected version for optimistic concurrency
+    aggregate.ExpectedVersion = events.Count();
+    
+    return aggregate;
+}
+```
 
 ### TakeEvents
 
-Takes the recorded history of events from this aggregate.
+Takes the recorded history of events from this aggregate. This method is typically called by a repository when saving the aggregate to extract the new events that need to be persisted.
```

Comment:
`TakeEvents` is also commonly used in unit tests of aggregates to validate that the correct events are raised.